spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/config/KinesisBinderConfiguration.java

```diff
@@ -67,14 +72,44 @@ public KinesisStreamProvisioner provisioningProvider(AmazonKinesisAsync amazonKi
 	KinesisMessageChannelBinder kinesisMessageChannelBinder(
 			AmazonKinesisAsync amazonKinesis,
 			KinesisStreamProvisioner provisioningProvider,
-			Codec codec) {
+			Codec codec, MetadataStore kinesisCheckpointStore) {
 
 		KinesisMessageChannelBinder kinesisMessageChannelBinder =
 				new KinesisMessageChannelBinder(amazonKinesis, this.configurationProperties, provisioningProvider);
 		kinesisMessageChannelBinder.setCodec(codec);
+		kinesisMessageChannelBinder.setMetaDataStore(kinesisCheckpointStore);
 
 		return kinesisMessageChannelBinder;
 	}
 
+	
+	@Bean
+	public AmazonDynamoDBAsync dynamoDB(AWSCredentialsProvider awsCredentialsProvider,
```

Comment:
I wouldn't provide a bean for the `AmazonDynamoDBAsync`. When end-user decides to override `DynamoDbMetaDataStore` bean, we will have some dangling bean for nothing.